lib/rucio/core/replica.py

```diff
@@ -2934,19 +2946,20 @@ def get_suspicious_files(rse_expression, **kwargs):
 
 
 @transactional_session
-def set_tombstone(rse_id, scope, name, session=None):
+def set_tombstone(rse_id, scope, name, tombstone=OBSOLETE, session=None):
     """
     Sets a tombstone on a replica.
 
     :param rse_id: ID of RSE.
     :param scope: scope of the replica DID.
     :param name: name of the replica DID.
+    :param tombstone: the tombstone to set. Default is OBSOLETE
     :param session: database session in use.
     """
     conn = get_engine().connect()
```

Comment:
not part of this patch but `conn = get_engine().connect()` ? It should be handled by the `transactional_session` decorator ? this can potentially lead to connection leak or abnormal transaction behaviour.

Reply:
Yes, that part looks strange indeed. If the engine object is needed, we should try to resolve this from the session directly. I will add a separate issue for this.

Reply:
#3071